source/grid/tria_accessor.cc

```diff
@@ -1531,6 +1531,17 @@ CellAccessor<dim, spacedim>::set_direction_flag (const bool new_direction_flag)
 
 template <int dim, int spacedim>
 void
+CellAccessor<dim, spacedim>::set_active_cell_index (const unsigned int active_cell_index)
+{
+  Assert (this->used(), TriaAccessorExceptions::ExcCellNotUsed());
```

Comment:
This function is not public, but only called from the Triangulation, and it is called for all cells (active or not) so as to reset the value for non-active cells. So I can't check whether the cell is active -- I want to specifically allow calling it for non-active ones as well.